src/reducers/filter.js

```diff
@@ -0,0 +1,16 @@
+import { CHANGE_FILTER } from "../actions";
+
+const initialState = {
+    filter: -1
+}
+
+
+export const filter = (state = initialState, action) => {
+    switch (action.type) {
+        case CHANGE_FILTER:
+            return {...state, ...{filter: action.data}};
```

Comment:
Just follow up clarification on this.. Why did we have to make our filter state as such:

![image](https://user-images.githubusercontent.com/6015315/57806635-215a3000-7792-11e9-90a5-4fb00594909e.png)

You may just make your state the integer value of `filte` instead of a full object like you did, such as:

```
const initialState = -1
```

Then your reducer can simply change that numerical filter value, for instance in the case for CHANGE_FILTER:

```
return {...state, action.data }
```

Hence, your store will simply be:

```
bookstore: {books: [array of books]},
filter: -1 (or whatever integer value)
```

While we are at this, why not just also directly make the array the value for your state.bookStore instead of nesting it inside another object? So that your improved store will be:

```
bookstore: [array of books],
filter: -1 (or whatever integer value)
```